fix: fixed CalendarGrid styling

knownotunknown opened the pull request:
fixed CalendarGrid styling
fixed unlimited rendering by removing dependency array with values changed inside useEffect
refactored PopupMain import to use existing switchSchedule method in lib directory. Deleted duplicate.

I should've made a separate branch and pull request, but I also refactored calendar schedules to use the activeScheduleIndex that was already managed by useSchedules. Also, I changed the order of how state was managed within the useSchedule hook because there was the possibility of stale data being re-used.

Linting and stuff now.

knownotunknown commented:
My bad, meant to make two different pull requests. Initially got side-tracked on the fixing-calendar-styles-branch, and then didn't realize I needed to make a new branch to actually fix calendar styling.

knownotunknown commented:
Also, CalendarGrid just needed to use useState with the useEffect function.

doprz commented on src/views/components/PopupMain.tsx:
```diff
@@ -8,10 +8,11 @@ import List from '@views/components/common/List/List';
 import { openTabFromContentScript } from '@views/lib/openNewTabFromContentScript';
 import clsx from 'clsx';
 import React, { useEffect, useRef, useState } from 'react';
+import switchSchedule from 'src/pages/background/lib/switchSchedule';
```
nit
Fix this import to follow ESLint `import-essentials/check-path-alias` please.

knownotunknown replied:
Fixed.

knownotunknown replied:
I don't need to worry about Chromatic failing though right?

doprz commented on src/views/hooks/useSchedules.ts:
```diff
@@ -42,18 +42,7 @@ export default function useSchedules(): [active: UserSchedule | null, schedules:
     return [activeSchedule, schedules];
-}
-
-/**
- * Switches the active schedule to the one with the specified name.
- * @param name - The name of the schedule to switch to.
- * @returns A promise that resolves when the active schedule has been switched.
- */
-export async function switchSchedule(name: string): Promise<void> {
-    const schedules = await UserScheduleStore.get('schedules');
-    const activeIndex = schedules.findIndex(s => s.name === name);
-    await UserScheduleStore.set('activeIndex', activeIndex);
-}
```
Why is this removed?

knownotunknown replied:
Meant to leave a message. The switchSchedule was already a method in the lib folder.

doprz reviewed (changes requested):
Please fix checks as all of them should be passing before merging.

knownotunknown commented:
fixed CalendarGrid styling 
fixed unlimited rendering by removing dependency array with values changed inside useEffect
refactored PopupMain import to use existing switchSchedule method in lib directory. Deleted duplicate.

doprz commented on src/views/components/calendar/CalendarGrid/CalendarGrid.tsx:
```diff
@@ -4,11 +4,11 @@ import { getCourseColors } from '@shared/util/colors';
 import styles from './CalendarGrid.module.scss';
 
-const daysOfWeek = Object.keys(DAY_MAP).filter(key => !['S', 'SU'].includes(key));
+const calendarGridDays = ['MON', 'TUE', 'WED', 'THU', 'FRI'];
```
👍

doprz commented on src/views/components/common/List/List.tsx:
```diff
@@ -60,6 +60,7 @@ function Item({ provided, item, style, isDragging /* , gap */ }) {
 export default function List({ draggableElements, gap = 12 }: ListProps): JSX.Element {
     const [items, setItems] = useState(() => initial(0, draggableElements));
 
+    // TODO: handle deleting schedules in useEffect function while maintaining previous order
```
Thanks for labeling this as a TODO

doprz commented on src/views/hooks/useSchedules.ts:
```diff
@@ -48,16 +51,5 @@ export default function useSchedules(): [active: UserSchedule | null, schedules:
-    const activeIndex = schedules.findIndex(s => s.name === name);
-    await UserScheduleStore.set('activeIndex', activeIndex);
+    return [activeSchedule, schedules, activeIndex];
 }
```
Thank you for removing this as it's now a util.

knownotunknown replied:
It was always a util. It was a util before that method was written.

doprz reviewed (changes requested):
There are some merge conflicts that need to be fixed before merging but great PR!

sghsri commented on src/views/hooks/useSchedules.ts:
```diff
@@ -17,23 +18,25 @@ export default function useSchedules(): [active: UserSchedule | null, schedules:
-
         fetchData();
 
         const setupListeners = () => {
             const l1 = UserScheduleStore.listen('schedules', ({ newValue }) => {
                 setSchedules(newValue.map(s => new UserSchedule(s)));
                 setActiveSchedule(currentActive => {
                     const newActiveIndex = newValue.findIndex(s => s.name === currentActive?.name);
+                    setActiveIndex(newActiveIndex);
                     return new UserSchedule(newValue[newActiveIndex]);
                 });
             });
 
             const l2 = UserScheduleStore.listen('activeIndex', ({ newValue }) => {
                 setSchedules(currentSchedules => {
                     setActiveSchedule(new UserSchedule(currentSchedules[newValue]));
+                    setActiveIndex(newValue);
                     return currentSchedules;
                 });
             });
```
make sure to remove the listeners when the useEffect destructor runs! 

```js
useEffect(() => {
	const l2 = UserScheduleStore.listen("activeIndex", ...
	return () => {
		UserScheduleStore.removeListener(l2);
	},
[]);
```

doprz replied:
Good catch
